src/mapGenerator/mapgenManager.py

```diff
@@ -0,0 +1,156 @@
+
+    def generateMap(self, mapname):
+        dataString =  mapname.split("neroxis_map_generator",1)[1]
+        version = re.search(r'\_(.*?)\_',dataString).group(1)
+        seed = dataString.rsplit('_', 1)[1]
+
+        actualPath = self.versionController(version)
+
+        if actualPath:
+            auto = Settings.get('mapGenerator/autostart', default=False, type=bool)
+            if not auto:
+                msgbox = QtWidgets.QMessageBox()
+                msgbox.setWindowTitle("Generate map")
+                msgbox.setText("Seems that you don't have the map used this game. Do you want to generate it?<br/><b>" + mapname + "</b>")
+                msgbox.setInformativeText("Map generation is CPU intensive task and may take some time.")
+                msgbox.setStandardButtons(QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.YesToAll | QtWidgets.QMessageBox.No)
+                result = msgbox.exec_()
+                if result == QtWidgets.QMessageBox.No:
+                    return False
+                elif result == QtWidgets.QMessageBox.YesToAll:
+                    Settings.set('mapGenerator/autostart', True)
+            
+            # Start generator with progress bar      
+            self.generatorProcess = MapGeneratorProcess(actualPath, self.mapsFolder, seed, version, mapname)
+            
+            # Check if map exists or generator failed
+            if os.path.isdir(os.path.join(self.mapsFolder, mapname)):
+                if self.previousMaps:
+                    self.previousMaps = self.previousMaps + ";" + mapname
+                else:
+                    self.previousMaps = mapname
+                Settings.set('mapGenerator/deleteMaps', self.previousMaps)
+                
+                return mapname
+            else:
+                return False
+        else:
+            return False
+            
+    def generateRandomMap(self):
+        ''' 
+        Called when user click "generate map" in host widget.
+        Prepares seed and requests latest version once per session
+        '''
+
+        if self.currentVersion == "0" or not self.latestVersion:
+            self.checkUpdates()
+
+            if self.latestVersion and self.versionController(self.latestVersion):
+                self.currentVersion = self.latestVersion   # mapgen is up-to-date
+                Settings.set('mapGenerator/version', self.currentVersion)
+            elif self.currentVersion == "0":               # if not "0", use older version
+                return False                               # otherwise we don't have any generator at all
+
+        seed = ''.join(["%s" % random.randint(0, 9) for num in range(0, 19)])
+        mapName = "neroxis_map_generator_{}_{}".format(self.currentVersion, seed)
+
+        return self.generateMap(mapName)
+   
+    def versionController(self, version):
+        name = "NeroxisGen_{}.jar".format(version)
+        filePath = self.generatorPath + "\\" + name
```

Comment:
NOOOOOOOOOOOOOOOOOOOOOOO!
JUST NO!
Never ever concatinate file paths manually.
First of all, this will break on any non Windows based operating system. Therefore you should use a constant/function provided by your programming language that gives you the OS specific path separator. (which for URLs is `/`, getting mad at you windows!)

Secondly, for this case you should always use a join/resolve function that gives you a file/directory inside your path, in python this is provided by os.path.join() I think.

Reply:
+1, use os.path.join.